primer-service/src/Primer/Servant/OpenAPI.hs

```diff
@@ -98,6 +98,13 @@ data SessionAPI mode = SessionAPI
           :> Get '[JSON] Prog
   , getSessionName :: GetSessionName mode
   , setSessionName :: SetSessionName mode
+    setSelection ::
```

Comment:
> we could common this up with the "get available actions" api

I don't really see how that would work?

> or split into a "set selection" and "get current node's type"

This is more like what I would have expected. Although I don't know if there are strong arguments either way, and I'm happy to revisit it later. I was actually expecting that we'd instead take the selection as an argument, rather than doing anything stateful at all. We have previously considered keeping track of all selections in the backend and decided against it: https://github.com/hackworthltd/primer/pull/706.